tycho-common/src/dto.rs

```diff
@@ -1802,14 +1802,91 @@ impl From<models::blockchain::AddressStorageLocation> for AddressStorageLocation
     }
 }
 
-#[derive(Serialize, Deserialize, Debug, Default, PartialEq, ToSchema, Eq, Clone)]
+fn deserialize_retriggers_from_value(
+    value: &serde_json::Value,
+) -> Result<HashSet<(StoreKey, AddressStorageLocation)>, String> {
+    use serde::Deserialize;
+    use serde_json::Value;
+
+    let mut result = HashSet::new();
+
+    if let Value::Array(items) = value {
+        for item in items {
+            if let Value::Array(pair) = item {
+                if pair.len() == 2 {
+                    let key = StoreKey::deserialize(&pair[0])
+                        .map_err(|e| format!("Failed to deserialize key: {}", e))?;
+
+                    // Handle both old format (string) and new format (AddressStorageLocation)
+                    let addr_storage = match &pair[1] {
+                        Value::String(_) => {
+                            // Old format: just a string key with offset defaulted to 0
+                            let storage_key = StoreKey::deserialize(&pair[1]).map_err(|e| {
+                                format!("Failed to deserialize old format storage key: {}", e)
+                            })?;
+                            AddressStorageLocation::new(storage_key, 0)
```

Comment:
I would suggest to make the default offset 12, even though it's ethereum specific.